### DISCUSSION TITLE
Using canvas in next.js, adding local fonts to canvas, but unable to find fonts after deployment is complete

### DISCUSSION THREAD

**Discussion by Potato-to** [vercel/next.js#64288] [Help] [answered]
### Summary

Add canvas in nextjs
I put it in the API to create images for use
**src and pubilc are of the same level**
// src/app/api/code/route.ts
```typescript
import { registerFont, deregisterAllFonts, createCanvas } from 'canvas';
deregisterAllFonts();
// Candaral.ttf  path pubilc/fonts/Candaral.ttf
registerFont('./public/fonts/Candaral.ttf', {family: 'registerFs'});
...
const context: any = canvas.getContext('2d');
context.font = 'aaa'+ ' ' + '"registerFs"';
const imageSrc = canvas.toDataURL('image/png');
return imageSrc;
```
When I run npm dev/start, it can be displayed
![image](https://github.com/vercel/next.js/assets/102496413/9f297960-07e6-4b42-a9cc-acfcc07ba46c)

When I deploy to Vercel, the page I open is blank
Error reported
```
Error: ENOENT: no such file or directory, lstat '/var/task/public'
    at Object.realpathSync (node:fs:2711:29)
    at registerFont (/var/task/node_modules/canvas/index.js:48:34)
    at 27494 (/var/task/.next/server/app/api/code/route.js:1:557)
    at t (/var/task/.next/server/webpack-runtime.js:1:143)
    at a (/var/task/.next/server/app/api/code/route.js:1:3117)
    at /var/task/.next/server/app/api/code/route.js:1:3150
    at t.X (/var/task/.next/server/webpack-runtime.js:1:1285)
    at /var/task/.next/server/app/api/code/route.js:1:3130
    at Object.<anonymous> (/var/task/.next/server/app/api/code/route.js:1:3178)
    at Module._compile (node:internal/modules/cjs/loader:1376:14) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/var/task/public',
  page: '/api/code'
}
```
Unable to display on my page
How should I solve it!

### Additional information

```js
"canvas": "^2.10.2",
"next": "^14.0.4",
```

**Comment by Potato-to** [comment, marked as answer]
The solution has been found   [https://github.com/orgs/vercel/discussions/4186](url)
I fixed the issue by adding outputFileTracingIncludes option in next.config.js.
Here is the code I used:
//  next.config.js
```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
...
  experimental: {
   outputFileTracingIncludes: {
      './src/app/api/code/route': ['./public/fonts/**/*'],
    },
  },
}
module.exports = nextConfig
```
I hope this helps some other programmers having the same issues.

**Reply by InnerDesires** [reply]
how do you reference the font files after this fix?

**Reply by gagandt** [reply]
Do you have a solution on referening the font files @InnerDesires @Potato-to 

I'm trying to reference the font added in my canvas, but my deployed API can't find it.

**Reply by Potato-to** [reply]
@gagandt Sorry for the late reply.
Take a look at my configuration outputFileTracingIncludes: {
'./src/app/api/code/route': ['./public/fonts/**/*'],
},
Here key is the position of the font currently needed.
Value is where you store fonts.
It's a little too long ago. I don't remember much.

**Reply by Potato-to** [reply]
@InnerDesires 
I looked at my code and it was introduced like this.
registerFont(path.join(process.cwd(), './public/fonts/CustomFont.ttf'), {family: 'CustomFont'});

**Comment by muhamadd9** [comment]
im watching
